examples/lighting-app/silabs/openthread.gn

```diff
@@ -24,6 +24,7 @@ default_args = {
   target_cpu = "arm"
   target_os = "freertos"
   chip_openthread_ftd = true
+  optimize_debug_level = "s"
```

Comment:
Why this change?

Reply:
I suggested this change to Terence because the lighting-app on MG12 was failing CI/running out of flash with this PR

Reply:
CI is failing due to space. Junior from Silabs mentioned over slack "(61a/ MG12 board) are starting to be tight in space", and "If it busts with your small changes it will bust for others soon"

I am the straw breaking the camels back. While I would like this to go in as a separate PR, because of code freeze I am trying to just get something in

Reply:
https://github.com/project-chip/connectedhomeip/pull/31555

Is in a standalond PR, plus now have a new idea to save some space